### ISSUE TITLE
how to exclude files using mypy.ini?

### ISSUE THREAD

**Issue by nagyben** [microsoft/vscode-mypy#157]
I've defined an exclude pattern in `mypy.ini` to exclude test files

```ini
#mypy.ini
[mypy]
warn_return_any = True
warn_unused_configs = True
exclude = (?x)(test_.+py)
...
```

This works fine when I run mypy from the command line in the terminal e.g. `mypy --config-file mypy.ini .` but because the plugin runs mypy on specific files (the output window for the plugin runs it like `python -m mypy ... [file_name]`) this exclude pattern is ignored

Setting `mypy-type-checker.args` also doesn't work:

```json
# settings.json
"mypy-type-checker.args": [
    "--exclude",
    "test"
]
```

What's the best way to exclude files?

**Comment by Mutantpenguin** [comment]
I have the same problem. With the "old" (and soon to be deprecated) way of running mypy (any pylint) you got to specify
```
"python.linting.ignorePatterns": [
			"**/site-packages/**/*.py",
			".vscode/*.py",
			"whatever/**/*.py"
		],
```
and I think the linter didn't even get invoked for files matching these patterns.

Mypy itself states in its documentation:
> Note that this flag only affects recursive directory tree discovery, that is, when mypy is discovering files within a directory tree or submodules of a package to check. If you pass a file or module explicitly it will still be checked. For instance, mypy --exclude '/setup.py$' but_still_check/setup.py.

See here for more information: https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-exclude

So in essence, I would like the "old" setting to work for linting extensions like this (and pylint) to work too. Even from a performance standpoint it should be faster (ignoring dmypy here, since it doesn't support "silent" it is of no use for me anyway right now).

**Comment by karthiknadig** [comment]
There is `mypy-type-checker.reportingScope` experiment setting that allows you to change the scope to `workspace`. The intent with that mode is to allow you to run linter over the workspace and a have the linter exclude any files. This is available in the pre-release. It is experimental, please try it and let me know if you see any issues.

**Comment by nagyben** [comment]
Adding 

```
"mypy-type-checker.reportingScope": "workspace"
```

to vscode settings solves this problem in my case

Thanks for the tip!

**Comment by MetRonnie** [comment]
It seems you need to do **all** of these in order for the exclude pattern in `mypy.ini` to be successfully picked up?
- Ensure `"mypy-type-checker.preferDaemon": true`
- Set `"mypy-type-checker.reportingScope": "workspace"`
- Set `"mypy-type-checker.args": [ "--config-file=${workspaceFolder}/mypy.ini" ]`
- Restart VSCode

This is painfully finicky

**Comment by Mutantpenguin** [comment]
@MetRonnie This has nothing to do with this vscode extension. Look here for an explanation https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-exclude

With the next update a fix by me will be included which allows you to set an ignore pattern in your *.code-workspace
https://github.com/microsoft/vscode-mypy/issues/213

You then don't need to set `mypy-type-checker.preferDaemon` and `mypy-type-checker.reportingScope` anymore.

**Comment by MetRonnie** [comment]
@Mutantpenguin I would hope that the extension would automatically use `mypy.ini` (or `pyproject.toml`) and not have to specify the excludes separately for VSCode unless you specifically wanted different excludes.

The first 2 bullet points are fair enough:

> * Ensure `"mypy-type-checker.preferDaemon": true`
> * Set `"mypy-type-checker.reportingScope": "workspace"`

But the latter two are what caused me pain to figure out

**Comment by Mutantpenguin** [comment]
I don't think that this extension should consider these files as it is only a wrapper around mypy, but this is not on me to decide as I am not a developer of it.

I guess what you need would be for mypy itself to honor the exlude settings in mypy.ini when it's directly invoked for a file. But this would need an issue for mypy itself and not this extension.

**Comment by jyates-om1** [comment]
Thanks for the call-out to `"mypy-type-checker.reportingScope": "workspace"` -> we are using mypy with configuration in a `pyproject.toml` and this was the missing component to get vscode to ignore our test code. Now our devcontainer behavior matches our type checker in our CI pipeline, which is great.
